File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/util/PlainTextSpans.java

package io.github.abc15018045126.sora.lang.util;

import io.github.abc15018045126.sora.lang.styling.EmptyReader;
import io.github.abc15018045126.sora.lang.styling.Spans;
import io.github.abc15018045126.sora.text.CharPosition;

/**
 * {@link Spans} implementation that always returns {@link EmptyReader} for reading spans.
 * Line count is automatically adjusted as content changes.
 *
 * @author abc15018045126
 */
public class PlainTextSpans implements Spans {

    private int lineCount;

    public PlainTextSpans(int lineCount) {
        this.lineCount = lineCount;
    }

    public void setLineCount(int lineCount) {
        this.lineCount = lineCount;
    }

    @Override
    public void adjustOnInsert(CharPosition start, CharPosition end) {
        lineCount += end.line - start.line;
    }

    @Override
    public void adjustOnDelete(CharPosition start, CharPosition end) {
        lineCount -= end.line - start.line;
    }

    @Override
    public Reader read() {
        return EmptyReader.INSTANCE;
    }

    @Override
    public boolean supportsModify() {
        return false;
    }

    @Override
    public Modifier modify() {
        throw new UnsupportedOperationException();
    }

    @Override
    public int getLineCount() {
        return lineCount;
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/util/PlainTextSpans.kt
package io.github.abc15018045126.sora.lang.util

import io.github.abc15018045126.sora.lang.styling.EmptyReader
import io.github.abc15018045126.sora.lang.styling.Spans
import io.github.abc15018045126.sora.text.CharPosition

/**
 * [Spans] implementation that always returns [EmptyReader] for reading spans.
 * Line count is automatically adjusted as content changes.
 *
 * @author abc15018045126
 */
class PlainTextSpans(private var lineCount: Int) : Spans {

    fun setLineCount(lineCount: Int) {
        this.lineCount = lineCount
    }

    override fun adjustOnInsert(start: CharPosition, end: CharPosition) {
        lineCount += end.line - start.line
    }

    override fun adjustOnDelete(start: CharPosition, end: CharPosition) {
        lineCount -= end.line - start.line
    }

    override fun read(): Spans.Reader {
        return EmptyReader.INSTANCE
    }

    override fun supportsModify(): Boolean {
        return false
    }

    override fun modify(): Spans.Modifier {
        throw UnsupportedOperationException()
    }

    override fun getLineCount(): Int {
        return lineCount
    }
}
